catalog/app/containers/Bucket/PackageUpdateDialog.js

```diff
@@ -0,0 +1,933 @@
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+}))
+
+function DialogError({ error, close }) {
+  const classes = useDialogErrorStyles()
+  return (
+    <>
+      <M.DialogTitle>Push package revision</M.DialogTitle>
+      <M.DialogContent style={{ paddingTop: 0, position: 'relative' }}>
+        <PD.FormSkeleton animate={false} />
+        <div className={classes.overlay}>{errorDisplay(error)}</div>
+      </M.DialogContent>
+      <M.DialogActions>
+        <M.Button onClick={close}>Cancel</M.Button>
+        <M.Button variant="contained" color="primary" disabled>
+          Push
+        </M.Button>
+      </M.DialogActions>
+    </>
+  )
+}
+
+function DialogSuccess({ bucket, name, revision, close }) {
+  const { urls } = NamedRoutes.use()
+  return (
+    <>
+      <M.DialogTitle>Push complete</M.DialogTitle>
+      <M.DialogContent style={{ paddingTop: 0 }}>
+        <M.Typography>
+          Package revision{' '}
+          <StyledLink to={urls.bucketPackageTree(bucket, name, revision)}>
+            {name}@{revision}
+          </StyledLink>{' '}
+          successfully created
+        </M.Typography>
+      </M.DialogContent>
+      <M.DialogActions>
+        <M.Button onClick={close}>Close</M.Button>
+        <M.Button
+          component={Link}
+          to={urls.bucketPackageTree(bucket, name, revision)}
+          variant="contained"
+          color="primary"
+        >
+          Browse
+        </M.Button>
+      </M.DialogActions>
+    </>
+  )
+}
+
+const DialogState = tagged([
+  'Closed',
+  'Loading',
+  'Error',
+  'Form', // { manifest, workflowsConfig }
+  'Success', // { name, revision }
+])
+
+export function usePackageUpdateDialog({ bucket, name, revision, onExited }) {
+  const s3 = AWS.S3.use()
+
+  const [isOpen, setOpen] = React.useState(false)
+  const [wasOpened, setWasOpened] = React.useState(false)
+  const [exited, setExited] = React.useState(!isOpen)
+  const [exitValue, setExitValue] = React.useState(null)
+  const [success, setSuccess] = React.useState(false)
+  const [key, setKey] = React.useState(1)
+
+  const manifestData = Data.use(
+    requests.loadManifest,
+    { s3, bucket, name, revision, key },
+    { noAutoFetch: !wasOpened },
+  )
+  const workflowsData = Data.use(requests.workflowsList, { s3, bucket })
+
+  const open = React.useCallback(() => {
+    setOpen(true)
+    setWasOpened(true)
+    setExited(false)
+  }, [setOpen, setWasOpened, setExited])
+
+  const close = React.useCallback(() => {
+    setOpen(false)
+    setSuccess(false)
+    setExitValue({ pushed: success })
+  }, [setOpen, setSuccess, success, setExitValue])
+
+  const refreshManifest = React.useCallback(() => {
+    setWasOpened(false)
+    setKey(R.inc)
+  }, [setWasOpened, setKey])
+
+  const handleExited = React.useCallback(() => {
+    setExited(true)
+    setExitValue(null)
+    if (onExited) {
+      const shouldRefreshManifest = onExited(exitValue)
+      if (shouldRefreshManifest) refreshManifest()
```

Comment:
I didn't follow, how can I refresh manifest. What actions in a browser should I make?

Reply:
manifest is refetched when you push a new revision while viewing the "latest" revision of the package

Reply:
Hmm, `shouldRefreshManifest` is always `undefined` or `false` for me. I couldn't find a case with `true`

Reply:
Found that case, finally:

1. `PackageRevisions` page
2. Push new revision
3. **Close "Push complete" dialog**